sklearn/decomposition/dict_learning.py

```diff
@@ -936,20 +1164,37 @@ class DictionaryLearning(BaseEstimator, SparseCodingMixin):
         number of dictionary elements to extract
 
     alpha : float,
-        sparsity controlling parameter
+        sparsity controlling parameter for L1 dictionary learning methods.
 
     max_iter : int,
         maximum number of iterations to perform
 
     tol : float,
         tolerance for numerical error
 
-    fit_algorithm : {'lars', 'cd'}
-        lars: uses the least angle regression method to solve the lasso problem
-        (linear_model.lars_path)
-        cd: uses the coordinate descent method to compute the
-        Lasso solution (linear_model.Lasso). Lars will be faster if
-        the estimated components are sparse.
+    n_nonzero_coefs: int, 0.1 * n_features by default
+        number of nonzero coefficients to target in each column of the
+        solution. This is used by L0 dictionary learning methods.
+
+    fit_algorithm : {'lars', 'cd', 'ksvd', 'exact_ksvd'}
+        Method for dictionary learning.
+        lars : update dictionary using block component wise
+            coordinate descent and the least angle regression method
+            to solve the lasso problem (linear_model.lars_path) for
+            sparse coding.
+        cd: update dictionary using block component wise
+            coordinate descent and the coordinate descent method to
+            compute the Lasso solution (linear_model.Lasso) for sparse
+            coding. Lars will be faster if the estimated components are
+            sparse.
+        ksvd : update atoms separately using approximate 1-svd
+            error matrix decomposition; in the most cases this
+            method is more preferable than the 'exact_ksvd' as it
+            gives good precision but works faster. The sparse coder
+            for learning is OMP.
+        exact_ksvd : update atoms separately using exact 1-svd
+            error matrix decomposition. The sparse coder for learning
+            is OMP.
```

Comment:
AFAIK it is not commonly used, but it was easy to implement, so I did it just in case. I can remove it if you want.